### DISCUSSION TITLE
How can I persist the user of rabbitmq

### DISCUSSION THREAD

**Discussion by ghost** [docker-library/rabbitmq#605] [Q&A]
So, I'm have a question, about de Rabbitmq. I'm getting persist data of   my project rabbitmq , but when i put my docker down, and docker up, all things persist but the user no. Some one can help me?

**Comment by michaelklishin** [comment]
Use a persistent volume for node's data directory. It is at [`/var/lib/rabbitmq`](https://github.com/docker-library/rabbitmq/blob/master/3.11/ubuntu/Dockerfile#L170) by default.

**Comment by lukebakken** [comment]
https://docs.docker.com/storage/volumes/

**Comment by ghost** [comment]
Hey, so I'm using this volume default, but when I make a "docker compose up" make a login on rabbitmq and add a new user, the rabbitmq save my user but, when I make "docker compose down" and "docker compose up" my users don't be save. However the all other things it's save.
Look this is my compose

  rabbitmq:
    image: rabbitmq:3-management-alpine
    container_name: rabbitmq
    command: "/bin/bash -c \"rabbitmq-plugins enable rabbitmq_mqtt; rabbitmq-server\""
    ports:
      - 5672:5672
      - 15672:15672
      - 1883:1883
    networks:
      - compose-bridge  
    environment:
      - RABBITMQ_DEFAULT_USER=admin
      - RABBITMQ_DEFAULT_PASS=pass123
    volumes:
      - ./rabbitmq/data/:/var/lib/rabbitmq/

**Reply by michaelklishin** [reply]
@YukioBunny we do not have much to add: your node's data store directory must be kept on a persistent volume. There is nothing else to do: this image does not use RAM nodes, so the entire schema data store (that includes virtual hosts, users, permissions, and so on) is stored on disk. Only if the data directory resides on a persistent volume.
